samples/download-preplanned-map-area/src/main/java/com/esri/arcgismaps/sample/downloadpreplannedmaparea/MainActivity.kt

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.downloadpreplannedmaparea
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.esri.arcgismaps.sample.downloadpreplannedmaparea.screens.DownloadPreplannedMapAreaScreen
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
+import java.io.File
+
+class MainActivity : ComponentActivity() {
+
+    private val offlineMapDirectory by lazy {
+        File(application.externalCacheDir?.path + application.getString(R.string.download_preplanned_map_area_app_name))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.ACCESS_TOKEN)
+
+        setContent {
+            SampleAppTheme {
+                DownloadPreplannedMapAreaApp()
+            }
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // Delete the offline map directory when the app is stopped
+        offlineMapDirectory.deleteRecursively()
+    }
```

Comment:
As an alternative, the compose approach would be to use a `DipsposableEffect`:
```kotlin
val context = LocalContext.current
DisposableEffect(Unit) {
    onDispose {
        File(context.externalCacheDir?.path + sampleName).let {
            if (it.exists()) { it.deleteRecursively() }
        }
    }
}
```

I'm more curious to understand, was there a reason why the cache directory contents are cleared `onDestroy`, and why not `onCreate`? If the sample's `onDestroy` is not called (on force stop/crash) then an error is shown when trying to download the preplanned map:

![image](https://github.com/user-attachments/assets/f6e057e3-5f5e-4a55-9ae8-3b3917237b10)

Reply:
Yeah good idea--I'll used onCreate which we know is guaranteed to run